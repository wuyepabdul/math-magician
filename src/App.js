import React from 'react';
import { HashRouter as Router, Switch, Route } from 'react-router-dom';
import NavBar from './components/NavBar';
import CalculatorPage from './pages/CalculatorPage';
import Home from './pages/Home';
import Quote from './pages/Quote';

const App = () => (
  <Router>
    <div className="container">
      <NavBar />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route path="/calculator">
          <CalculatorPage />
        </Route>
        <Route path="/quote">
          <Quote />
        </Route>
      </Switch>
    </div>
  </Router>
);

export default App;
