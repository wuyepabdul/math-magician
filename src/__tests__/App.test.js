import '@testing-library/jest-dom/extend-expect';
import { render, screen, cleanup } from '@testing-library/react';
import App from '../App';

describe('AppRouting and Navbar', () => {
  afterEach(() => {
    cleanup();
  });
  test('navbar is loaded and contains elements', () => {
    render(<App />);
    const navBar = screen.getByTestId('navbar');
    expect(navBar).toBeInTheDocument();
    expect(navBar.firstChild.textContent).toBe(' Math Magicians ');
    const navElements = navBar.childNodes[1];
    expect(navElements.classList).toContain('nav-links');
    expect(navElements.childNodes.length).toEqual(3);
    expect(navElements.childNodes[1].textContent).toEqual('Calculator');
  });
  test('home default', () => {
    render(<App />);
    const homeDefault = screen.getByTestId('home-div');
    expect(homeDefault).toBeInTheDocument();
    expect(homeDefault.firstChild.textContent).toEqual('Welcome to our page!');
  });
  test('snapshot of Whole APP Js page', () => {
    const mainAppComponent = render(<App />);
    expect(mainAppComponent).toMatchSnapshot();
  });
});
