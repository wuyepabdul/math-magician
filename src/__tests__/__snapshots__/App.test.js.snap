// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouting and Navbar snapshot of Whole APP Js page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <nav
          class="nav"
          data-testid="navbar"
        >
          <h1
            class="logo"
          >
             
            <a
              aria-current="page"
              class="active"
              href="#/"
            >
              Math Magicians
            </a>
             
          </h1>
          <ul
            class="nav-links"
          >
            <li>
              <a
                aria-current="page"
                class="active-link"
                href="#/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="#/calculator"
              >
                Calculator
              </a>
            </li>
            <li>
              <a
                href="#/quote"
              >
                Quote
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="home"
          data-testid="home-div"
        >
          <h1
            class="home-heading"
          >
            Welcome to our page!
          </h1>
          <p
            data-testid="p-tag"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ipsa nemo unde eaque. Id, porro. Vitae repudiandae officiis tempora, fugiat eligendi iusto nam, placeat vel expedita, aliquam ad ex provident ab.Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ipsa nemo unde eaque. Id, porro. Vitae repudiandae officiis tempora, fugiat eligendi iusto nam, placeat vel expedita, aliquam ad ex provident ab.
             
            <br />
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ipsa nemo unde eaque. Id, porro. Vitae repudiandae officiis tempora, fugiat eligendi iusto nam, placeat vel expedita, aliquam ad ex provident ab.Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ipsa nemo unde eaque. Id, porro. Vitae repudiandae officiis tempora, fugiat eligendi iusto nam, placeat vel expedita, aliquam ad ex provident ab.
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <nav
        class="nav"
        data-testid="navbar"
      >
        <h1
          class="logo"
        >
           
          <a
            aria-current="page"
            class="active"
            href="#/"
          >
            Math Magicians
          </a>
           
        </h1>
        <ul
          class="nav-links"
        >
          <li>
            <a
              aria-current="page"
              class="active-link"
              href="#/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="#/calculator"
            >
              Calculator
            </a>
          </li>
          <li>
            <a
              href="#/quote"
            >
              Quote
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="home"
        data-testid="home-div"
      >
        <h1
          class="home-heading"
        >
          Welcome to our page!
        </h1>
        <p
          data-testid="p-tag"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ipsa nemo unde eaque. Id, porro. Vitae repudiandae officiis tempora, fugiat eligendi iusto nam, placeat vel expedita, aliquam ad ex provident ab.Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ipsa nemo unde eaque. Id, porro. Vitae repudiandae officiis tempora, fugiat eligendi iusto nam, placeat vel expedita, aliquam ad ex provident ab.
           
          <br />
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ipsa nemo unde eaque. Id, porro. Vitae repudiandae officiis tempora, fugiat eligendi iusto nam, placeat vel expedita, aliquam ad ex provident ab.Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ipsa nemo unde eaque. Id, porro. Vitae repudiandae officiis tempora, fugiat eligendi iusto nam, placeat vel expedita, aliquam ad ex provident ab.
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
